website/addons/s3/tests/test_view.py

```diff
@@ -219,185 +121,63 @@ def test_user_settings_cant_list(self, mock_can_list):
-        url = self.project.api_url_for('s3_authorize_node')
-        cred = {
-            'access_key': fake.password(),
-        }
-        res = self.app.post_json(url, cred, auth=self.user.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-
-    @mock.patch('website.addons.s3.views.config.utils.can_list', return_value=False)
-    def test_s3_authorize_user_invalid(self, _):
-        url = self.project.api_url_for('s3_authorize_node')
-        cred = {
-            'access_key': fake.password(),
-            'secret_key': fake.password(),
-        }
-        res = self.app.post_json(url, cred, auth=self.user.auth, expect_errors=True)
-        assert_in('Unable to list buckets', res.json['message'])
-        assert_equal(res.status_code, 400)
-
-    def test_s3_authorize_input_empty_keys(self):
-        url = self.project.api_url_for('s3_authorize_node')
-        cred = {
-            'access_key': '',
-            'secret_key': '',
-        }
-        res = self.app.post_json(url, cred, auth=self.user.auth, expect_errors=True)
-        assert_in('All the fields above are required', res.json['message'])
-        assert_equal(res.status_code, 400)
-
-    def test_s3_authorize_input_empty_access_key(self):
-        url = self.project.api_url_for('s3_authorize_node')
-        cred = {
-            'access_key': '',
-            'secret_key': 'Non-empty-secret-key',
-        }
-        res = self.app.post_json(url, cred, auth=self.user.auth, expect_errors=True)
-        assert_in('All the fields above are required', res.json['message'])
-        assert_equal(res.status_code, 400)
-
-    def test_s3_authorize_input_empty_secret_key(self):
-        url = self.project.api_url_for('s3_authorize_node')
-        cred = {
-            'access_key': 'Non-empty-access-key',
-            'secret_key': '',
-        }
-        res = self.app.post_json(url, cred, auth=self.user.auth, expect_errors=True)
-        assert_in('All the fields above are required', res.json['message'])
-        assert_equal(res.status_code, 400)
-
-    @mock.patch('website.addons.s3.views.config.utils.can_list', return_value=True)
-    def test_s3_node_import_auth_authorized(self, _):
-        url = self.project.api_url_for('s3_node_import_auth')
-        self.node_settings.deauthorize(auth=None, save=True)
-        res = self.app.post(url, auth=self.user.auth)
-        assert_equal(res.json['node_has_auth'], True)
-        assert_equal(res.json['user_is_owner'], True)
-
-    def test_s3_node_import_auth_unauthorized(self):
-        url = self.project.api_url_for('s3_node_import_auth')
-        self.node_settings.deauthorize(auth=None, save=True)
-        unauthorized = AuthUserFactory()
-        res = self.app.post(url, auth=unauthorized.auth, expect_errors=True)
-        assert_equal(res.status_code, 403)
+    ## Overrides ##
+
+    @mock.patch('website.addons.s3.views.utils.get_bucket_names')
+    def test_folder_list(self, mock_names):
```

Comment:
It's overriding `addons.base.testing.views.OAuthAddonConfigViewsTestCaseMixin.test_folder_list` for the sake of mocking an outgoing API call, and then performing [that test](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/base/testing/views.py#L196-L205)